fe/fe-core/src/main/java/com/starrocks/journal/bdbje/BDBJEJournal.java

```diff
@@ -492,6 +498,8 @@ public void batchWriteAppend(long journalId, DataOutputBuffer buffer) throws Int
     /**
      * persist current batch
      * for bdb: commit current transaction
+     * notice that if commit fail, the transaction may not be valid.
+     * we should rebuild the transaction and retry.
      */
     @Override
     public void batchWriteCommit() throws InterruptedException, JournalException {
```

Comment:
Better to clear the uncommitedDatas and set currentTrasaction to null in this function no matter the result is success or fail. To avoid some inconsistent.